src/Elcodi/Component/Plugin/Templating/Traits/TemplatingTrait.php

```diff
@@ -68,4 +70,28 @@ protected function appendTemplate(
                 )
         );
     }
+
+    /**
+     * Plugin can be used
+     *
+     * @param Plugin $plugin          Plugin
+     * @param array  $checkableFields Fields to check
+     *
+     * @return boolean Plugin can be used
+     */
+    protected function pluginCanBeUsed(
+        Plugin $plugin,
+        array $checkableFields = []
+    ) {
+        return array_reduce(
+            $checkableFields,
+            function ($canBeUsed, $checkableField) use ($plugin) {
+
+                return $canBeUsed &&
+                    isset($plugin->getConfiguration()[$checkableField]) &&
```

Comment:
This is calling aloud for a `getSettings()` and a `getSetting($name)` methods in `PluginInterface`.